### PR TITLE
DATAGO-74500  | Added Container Security Context

### PR DESCRIPTION
**Pull request by LewisKSaint**

### CODE REVIEW

File: .github/workflows/build-test-dev.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
       - main
 
 env:
-  VERSION: 1.0.3
+  VERSION: 1.2.0-dev
```
Comment (hossein-ashraf-kashani) [inline .github/workflows/build-test-dev.yml:9]: what is the criteria for selecting the next version?

File: deploy/deploy.yaml [hunk 2]
```diff
@@ -1903,7 +1920,7 @@ spec:
         env:
         - name: WATCH_NAMESPACE
           value: ""
-        image: docker.io/solace/pubsubplus-eventbroker-operator:1.0.3
+        image: ghcr.io/solacedev/pubsubplus-eventbroker-operator:1.2.0
```
Comment (hossein-ashraf-kashani) [inline deploy/deploy.yaml:1923]: is this required as part of your change?

Comment (LewisKSaint) [reply deploy/deploy.yaml:1923]: Yeah, the release points to users to the dockerhub for the latest image but during development we use GHCR as the images change and we only push to dockerhub as part of the release.

File: .github/workflows/build-test-dev.yml (hunk 1 above)
Comment (LewisKSaint) [reply .github/workflows/build-test-dev.yml:9]: This is just for dev testing.

**Review by hossein-ashraf-kashani — APPROVED** [review]

**Pull request merged**